crates/dht/src/arc_set.rs

```diff
@@ -10,8 +10,7 @@ use kitsune2_api::{DhtArc, K2Error, K2Result};
 use std::collections::HashSet;
 
 /// Represents a set of [DhtArc]s.
-#[derive(Debug, Clone)]
-#[cfg_attr(test, derive(PartialEq))]
```

Comment:
This is still only used for tests elsewhere, I could introduce a `test-utils` feature but is it worth it?

Reply:
Ah no, that's fine.

Reply:
There's nothing incorrect about PartialEq for this type, and compiler tree-shaking should delete the code if it's not used in production. I'm happy for things like this to exist on our types.